aws-java-sdk-core/src/main/java/com/amazonaws/ClientConfiguration.java

```diff
@@ -474,12 +497,41 @@ public ClientConfiguration withProxyHost(String proxyHost) {
     }
 
     /**
-     * Returns the optional proxy port the client will connect through.
+     * Returns the Java system property for proxy port depending on
+     * {@link this.getProtocol()}: i.e. if protocol is https, returns
+     * the value of the system property https.proxyPort, otherwise
+     * returns value of http.proxyPort.
```

Comment:
Would be nice if we can have the javadoc updated with "Defaults to proxyPort is the System property is not set with a valid port number"